### ISSUE TITLE
ShouldOpen and ShouldClose delegate methods FABMenu

### ISSUE THREAD

**Issue by mjurenka** [CosmicMind/Material#1043]
Hello,

thanks for an awesome library for creating material apps!

I would like to add following methods to the FABMenuDelegate:

```
optional func fabMenuShouldOpen(fabMenu: FABMenu) -> Bool

optional func fabMenuShouldClose(fabMenu: FABMenu) -> Bool
```

The reason is simple: have some way to prevent opening/closing of the FAB menu when desirable.
I can provide PR.

Thanks!
Martin

**Comment by danieldahan** [comment]
Yes please, I would be more than happy to add such an addition based on your PR :) Thank you!

**Comment by OrkhanAlikhanov** [comment]
#1046 had been merged, so the next release gonna have this.

**Comment by danieldahan** [comment]
This has been released in [Material 2.16.0](https://github.com/CosmicMind/Material/releases/tag/2.16.0). Thank you!
